### ISSUE TITLE
Documentation for Linux "Targeted Usage" is incorrect

### ISSUE THREAD

**Issue by jasonsewall-intel** [intel/opencl-intercept-layer#73]
## Observed Behavior

Passing CLI_DLLName=<whatever> doesn't work, because Linux is case-sensitive and the code looks for CLI_DllName.

The documentation under install.md lists the upper-case version.

## Desired Behavior

Change the documentation to demonstrate the lower-case version (or perhaps change the preferred variable name to the upper-case version, or check for both).

## Steps to Reproduce

Set CLI_DLLName=<whatever>, and opencl-intercept-layer will not find it with getenv.

This should be an issue with any POSIX OS, regardless of application or other variables set. Having something set in ~/clintercept.conf might change the behavior, I haven't tried that.

Some of this information may automatically be included if you attach your Intercept Layer for OpenCL Applications log, particularly if you set [CLInfoLogging](https://github.com/intel/opencl-intercept-layer/blob/master/docs/controls.md#clinfologging-bool).)

**Comment by bashbaug** [comment]
Whoops, thanks, I'll fix this.

Do you think it would be better to make the controls case-insensitive instead?

**Comment by jasonsewall-intel** [comment]
Probably a lot of work to handle multiple conflicting entries that collide in a sane way; I'd recommend just pickng a single case (you are already ticking on the Linux folks by calling their SO files "DLLs" : )

**Comment by bashbaug** [comment]
I've fixed the docs in https://github.com/intel/opencl-intercept-layer/commit/dd9aaf6fa8a44ef3ddecf5025f9e523a4b4dcc2e.

I'll leave this issue open a bit longer to see if there's an easy way to make the controls case-insensitive, but feel free to close if it you'd like.

> you are already ticking on the Linux folks by calling their SO files "DLLs"

In all seriousness, this has been bugging me, but I haven't come up with a better equivalent, and I don't want to change it twice.  I'm open to suggestions!  Things I've considered include: `LibName`, `ICDLoaderName`, `RealICDLoader`, ...

**Comment by jasonsewall-intel** [comment]
It's a pain to support case-insensitivity through just getenv(), since I suppose you'd have to query all combinations of cases.  There may be other interfaces, but most of all, I'm worried about the logic of trying to decide what to do when someone has set DLLName=<thing1> and DllName=<thing2> and maybe dllname=<thing3>, which is unfortunately a very real possibility with environment variables.

So I say it should remain case-sensitive. 

As for the name of the library, the fact that you even have to say what library you are replacing is weird to me. I think in Linux at least, you can use dlsym(RTLD_NXT, "symbol") to get symbols from the object that ld would find *after* the current object. This means that if you used LD_PRELOAD or some LD_LIBARY_PATH trickery to put the intercept libOpenCL.so in front of the system one, you can still find symbols from the 'original' one.
I have no idea what you would do on Windows here.

**Comment by bashbaug** [comment]
> It's a pain to support case-insensitivity through just getenv(), since I suppose you'd have to query all combinations of cases.

I investigated making the controls case-insensitive, and in short, I agree that this would be very challenging for environment variables on Linux.  I think we could make it work for controls in config files, but I'm not sure that's worth it.  For the record, because of the way environment variables (and registry keys) work on Windows they'll get treated as case insensitive, for now at least, but I wouldn't consider it safe to rely on that behavior.

I've updated the docs to say that the controls should be treated as case sensitive with commit a5a1eadd3abe739c76949dc0842d06f4a0900df6, so I'm closing this issue.

Thanks again for the bug report!
